### PR TITLE
Creating a modal script that will download the entire wikipedia dataset and embed it

### PR DESCRIPTION
**Pull request by ivanleomk**

### CODE REVIEW

**Comment by jxnl** [comment]
can you add a title so i know the goal of this pr?

**Comment by jxnl** [comment]
i updated batch size.
can you make sure to  use `ruff` on save?

also best to move the saving to a function, lets use a dataclass to organize the response objects too

File: applications/wikipedia/main.py [hunk 1]
```diff
@@ -58,86 +68,208 @@ def download_model():
...
+        n_chars = sum(map(len, texts))
+        res = await self.client.post("/embed", json={"inputs": texts})
+        embeddings = res.json()
+        end = time.perf_counter()
+        total_time = int(end - start)
+        snapshot = self.gpu_utilization_figures[-total_time:]
+        if not snapshot:
+            snapshot = [get_gpu_utilization()[0]]
+        gpu_utilization = sum(snapshot) / len(snapshot)
+        return list(zip(texts, embeddings)), n_chars, gpu_utilization
```
Comment (jxnl) [inline applications/wikipedia/main.py:170]: zipping is extra overhead
texts exists in the main computer so zipping and sending it back is extra overhead.

File: applications/wikipedia/main.py [hunk 2]
```diff
@@ -58,86 +68,208 @@ def download_model():
...
+    async def embed(self, texts: list[str]):
+        start = time.perf_counter()
+        n_chars = sum(map(len, texts))
+        res = await self.client.post("/embed", json={"inputs": texts})
+        embeddings = res.json()
+        end = time.perf_counter()
+        total_time = int(end - start)
+        snapshot = self.gpu_utilization_figures[-total_time:]
+        if not snapshot:
+            snapshot = [get_gpu_utilization()[0]]
```
Comment (jxnl) [inline applications/wikipedia/main.py:168]: did you chech if this is expensive?

File: applications/wikipedia/main.py [hunk 3]
```diff
@@ -58,86 +68,208 @@ def download_model():
...
+    materialized_batchs = list(batches)
+    print(
+        f"Materialized {len(materialized_batchs)} batches in {time.perf_counter()-start:.2f} seconds"
+    )
+
+    start = time.perf_counter()
+    counter = 0
+    combined_embedding_and_text = []
+    gpu_utilization_snapshot = []
+    for embedding_and_text, n_chars, gpu_utilization in model.embed.map(
```
Comment (jxnl) [inline applications/wikipedia/main.py:223]: we can zip the map function to have access to texts

**Comment by jxnl** [comment]
https://docs.wandb.ai/guides/track/log/distributed-training#use-wb-service-to-avoid-common-distributed-training-issues

**Comment by ivanleomk** [comment]
Closing this PR because we're deprioritizing wandb support for now

**Pull request closed without merging**